varats-core/varats/project/project_util.py

```diff
@@ -57,25 +57,25 @@ def get_local_project_git_path(
     else:
         source = get_primary_project_source(project_name)
 
-    if hasattr(source, "fetch"):
+    if not isinstance(source, GitSubmodule) and hasattr(source, "fetch"):
```

Comment:
Why not fetch submodules here?

Also, use `is_git_source()` here:
```suggestion
    if not isinstance(source, GitSubmodule) and is_git_source(source):
```

Reply:
But why do we not fetch submodules? I would assume that if we get the git path to a submodule, we want to access it and hence, it should be fetched.

Reply:
Your question is related to one question of @vulder regarding the SOURCES list. I'll test if your proposed idea can solve both problems and let you know.

Reply:
The main problem is that we need the initialized submodules already when the main repo gets initialized. This is also the reason why I built the submodule list in the SOURCES before the VaraTestRepoSource. Considering that, I saw no reason to fetch() the submodules, at the referred code part, again.
I could not figure out how i can circumvent the non-optimal definition of SOURCES and the execution of code at class parse time, which @vulder already mentioned. But your proposed wrapper (VaraTestRepoSubmodule) for the GitSubmodules enables us to call fetch() on a submodule without an path-error, therefore i can remove the ```if not isinstance(source, GitSubmodule)``` part.

Reply:
Those are actually two different problems. 
For our test repo sources, we probably will not be able to get rid of this forward-reference.
This comment here and the wrapper around `GitSubmodule` are needed to preserve the submodule-abstraction built into benchbuild.